File: Client(Android)/app/src/main/java/com/example/integratedqueuesystem/data/agency/Agency.kt
package com.example.integratedqueuesystem.data.agency

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.integratedqueuesystem.data.reservation.ServiceType

@Entity(tableName = "agency_table")
data class AgencyData (
    @PrimaryKey
    val id: String,
    val name: String,
    val type: Int,
    val address: String,
    val latitude: Double,
    val longitude: Double,
) {
    constructor(agency: Agency) : this(
        agency.id,
        agency.name,
        agency.type,
        agency.address,
        agency.latitude,
        agency.longitude
    )
}

class Agency (
    val id: String = "00000",
    val name: String = "UnknownName",
    val type: Int = AgencyType.Unknown,
    val address: String = "UnknownAddress",
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    //val next_num: List<String> = emptyList(),
    val pending: List<Int> = emptyList(),//null?
    val status: Int = 0,
    //val queue: HashMap<String, ReservationData> = HashMap(),
    val dataFormat: List<ServiceType> = emptyList()
) {
    constructor(agencyData: AgencyData): this(
        agencyData.id,
        agencyData.name,
        agencyData.type,
        agencyData.address,
        agencyData.latitude,
        agencyData.longitude,
    )
}

@Entity(tableName = "favorite_agency_table")
data class FavoriteAgency (
    @PrimaryKey
    val agencyId: String,
)

File: Business(C#)/waitlinetest_frontandqueue/Firebase/Agency.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace waitlinetest_frontandqueue.Firebase
{
    public class Agency
    {
        public string address { get; set; }
        public Dictionary<string, int> datatype { get; set; }
        public string id { get; set; }
        public double latitude { get; set; }
        public double longitude { get; set; }
        public string name { get; set; }
        public string next_num { get; set; }
        public int pending { get; set; }
        public Dictionary<string, Data> queue { get; set; }
        public int status { get; set; }
        public int type { get; set; }
    }
}
